### PR TITLE
feat: sim replace frontend rpc calls with genlayerjs

### LINKED ISSUE #580: SIM-Replace frontend RPC calls with GenlayerJS

### PR DESCRIPTION
**Pull request by denishacquin**
Fixes #580 

# What

- Implemented GenLayer JS in the Studio frontend
- Added possibility to return null on **get_transaction_by_hash** endpoint (was breaking when a server-side tx didn't exist)

# Why

- To get closer to the actual final implementation and use the correct api's
- To fix a bug

# Testing done

- tested the new feature
- tested the bug fix

# Decisions made

- Removed some functionality that is now covered by the GenLayer JS library (nonces, useWallet hook...)

# Checks

- [x] I have tested this code
- [x] I have reviewed my own PR
- [x] I have created an issue for this PR
- [x] I have set a descriptive PR title compliant with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)

# User facing release notes

- Implemented the official GenLayer JS library to handle accounts and transactions in the Studio

### CODE REVIEW

**Comment by denishacquin** [comment]
@cristiam86 any hint on how to use **leaderonly** functionality with GenLayer JS?

**Comment by cristiam86** [comment]
> @cristiam86 any hint on how to use **leaderonly** functionality with GenLayer JS?

sure, to use it in GenLayerJS, it should be implemented there :D

**Comment by denishacquin** [comment]
Regarding the **leaderOnly** feature, I've been trying to implement it in genlayer-js, without success until now.

If I dare ask, how important is that feature? Won't it have to be removed once we hook the studio to the real chain? @cristiam86

**Comment by cristiam86** [comment]
> Regarding the **leaderOnly** feature, I've been trying to implement it in genlayer-js, without success until now.
> 
> If I dare ask, how important is that feature? Won't it have to be removed once we hook the studio to the real chain? @cristiam86

Let's take a look together whenever you can :) This is going to be available permanently in the localnet. Of course, it won't be available in the testnet or mainnet. 
In Ethereum and other L1s, there are some methods that can be only used in the localnet and they make it easy to force situations or to speed up the development.

File: frontend/src/hooks/useContractQueries.ts [hunk 1]
```diff
@@ -63,8 +63,9 @@ export function useContractQueries() {
       return mockContractSchema;
     }
 
-    const result = await rpcClient.getContractSchema({
-      code: contract.value?.content ?? '',
+    const result = await genlayer.client?.request({
```
[the remark this answers is not in the document]
Comment (denishacquin) [reply frontend/src/hooks/useContractQueries.ts:66]: Ah great! I think they weren't there yet when I started this integration! Will update asap!

**Comment by denishacquin** [comment]
Note that if we want to support the leaderOnly param in this PR, we need to merge the feature in genlayer-js then update the dependency here. @cristiam86

**Comment by denishacquin** [comment]
I am now having some typing errors between the genvm merge and genlayer-js when fetching contract schema, because the structure we get from the backend doesn't match the type defined in genlayer-js:

```export type ContractSchema = {
  abi: Array<{
    inputs?: Array<{
      name: string;
      type: string;
    }>;
    name?: string;
    outputs?: Array<{
      name: string;
      type: string;
    }>;
    type: "constructor" | "function";
  }>;
  class: string;
};
```
image

**Comment by cristiam86** [comment]
we just merged https://github.com/yeagerai/genlayer-studio/pull/653 which changes this as it should be. Backend returns bytes and the frontend decodes it. Now we need to do the same in GenLayerJS

**Comment by denishacquin** [comment]
> we just merged #653 which changes this as it should be. Backend returns bytes and the frontend decodes it. Now we need to do the same in GenLayerJS

From what I can see that is for eth_call, not for gen_getContractSchema / gen_getContractSchemaForCode

**Comment by cristiam86** [comment]
Oh, sure, sorry, my bad.
@kp2pml30 can we go ahead refactoring this in the frontend with the new schema?

**Comment by denishacquin** [comment]
Ok, then we simply need to update the ContractSchema type in genlayer-js to match the one in the studio.

**Comment by denishacquin** [comment]
Created an issue  theon genlayer-js side, will take care of it:
https://github.com/yeagerai/genlayer-js/issues/33

**Review by cristiam86 — CHANGES_REQUESTED** [review]
Great work, but we need to merge this one first: https://github.com/yeagerai/genlayer-js/pull/35 and update the leader_only parameter here.

**Comment by denishacquin** [comment]
> Great work, but we need to merge this one first: [yeagerai/genlayer-js#35](https://github.com/yeagerai/genlayer-js/pull/35) and update the leader_only parameter here.

Went ahead and merged but turns out we forgot a couple - fixed here: #6 please review @cristiam86 

As soon as it's released in 0.4.6 I'll bump the version here again!

**Review by denishacquin — COMMENTED** [review]
leader only was updated on both sides

**Review by cristiam86 — APPROVED** [review]

**Review by denishacquin — COMMENTED** [review]
leader only ok

**Pull request merged**